test/test_delegate.rb

```diff
@@ -131,6 +139,32 @@ def test_protected_instance_methods
     assert_equal([:parent_protected, :parent_protected_added], (Child.new(Parent.new).protected_methods - ignores).sort)
   end
 
+  def test_instance_methods
+    ignores = Object.instance_methods | Delegator.instance_methods
+    assert_equal([:parent_protected, :parent_protected_added, :parent_public, :parent_public_added], (Child.instance_methods - ignores).sort)
+    assert_equal([:parent_protected, :parent_protected_added, :parent_public, :parent_public_added], (Child.new(Parent.new).methods - ignores).sort)
+  end
+
+  def test_DelegateClass_instance_method
+    assert_instance_of UnboundMethod, Child.instance_method(:parent_public)
+    assert_instance_of UnboundMethod, Child.instance_method(:parent_public_added)
+    assert_instance_of UnboundMethod, Child.instance_method(:parent_protected)
+    assert_instance_of UnboundMethod, Child.instance_method(:parent_protected_added)
+    assert_raise(NameError) { Child.instance_method(:parent_private) }
+    assert_raise(NameError) { Child.instance_method(:parent_private_added) }
+    assert_instance_of UnboundMethod, Child.instance_method(:to_s)
```

Comment:
`instance_method` also raised an error when it is called to get `to_s` method.
So I also added the assertion.


```ruby
require 'delegate'

class Parent
  def parent() end
end

class Child < DelegateClass(Parent)
end

class Parent
  def parent_added() end
end

p Child.instance_method(:parent) # ok
p Child.instance_method(:to_s) # undefined method `to_s' for class `Child' (NameError)
```